PR: https://github.com/odoo/odoo/pull/45832

From: 3dffce34bf80c2a8d8682ef986d6daafa5bb0b7f
From: mreficent
Date: 2020-02-20 14:58:33

Structural Changes: 8
Total Changes: 12

[IMP] fields: log warning for ignored selection

When a field is related, defining a selection or selection_add will
have no effect and the paramater is ignored.

Log a warning and fix all fields badly definied

Closes odoo/odoo#45716

closes odoo/odoo#45832

Related: odoo/enterprise#8613
Signed-off-by: Martin Trigaux (mat) <mat@odoo.com>

================================= pseudo patch: =================================

--- a/addons/stock/models/stock_inventory.py
+++ b/addons/stock/models/stock_inventory.py
@@ -357,7 +357,7 @@ class InventoryLine(models.Model):
     company_id = fields.Many2one(
         'res.company', 'Company', related='inventory_id.company_id',
         index=True, readonly=True, store=True)
-    state = fields.Selection('Status', related='inventory_id.state')
+    state = fields.Selection(string='Status', related='inventory_id.state')
     theoretical_qty = fields.Float(
         'Theoretical Quantity',
         digits='Product Unit of Measure', readonly=True)
@@ -369,7 +369,7 @@ class InventoryLine(models.Model):
         help="Last date at which the On Hand Quantity has been computed.")
     outdated = fields.Boolean(string='Quantity outdated',
         compute='_compute_outdated', search='_search_outdated')
-    product_tracking = fields.Selection('Tracking', related='product_id.tracking', readonly=True)
+    product_tracking = fields.Selection(string='Tracking', related='product_id.tracking', readonly=True)
 
     @api.depends('product_qty', 'theoretical_qty')
     def _compute_difference(self):

--- a/addons/stock/models/stock_picking.py
+++ b/addons/stock/models/stock_picking.py
@@ -315,10 +315,8 @@ class Picking(models.Model):
         'stock.picking.type', 'Operation Type',
         required=True, readonly=True,
         states={'draft': [('readonly', False)]})
-    picking_type_code = fields.Selection([
-        ('incoming', 'Vendors'),
-        ('outgoing', 'Customers'),
-        ('internal', 'Internal')], related='picking_type_id.code',
+    picking_type_code = fields.Selection(
+        related='picking_type_id.code',
         readonly=True)
     picking_type_entire_packs = fields.Boolean(related='picking_type_id.show_entire_packs',
         readonly=True)

--- a/addons/stock/models/stock_scrap.py
+++ b/addons/stock/models/stock_scrap.py
@@ -36,7 +36,7 @@ class StockScrap(models.Model):
         'uom.uom', 'Unit of Measure',
         required=True, states={'done': [('readonly', True)]}, domain="[('category_id', '=', product_uom_category_id)]")
     product_uom_category_id = fields.Many2one(related='product_id.uom_id.category_id')
-    tracking = fields.Selection('Product Tracking', readonly=True, related="product_id.tracking")
+    tracking = fields.Selection(string='Product Tracking', readonly=True, related="product_id.tracking")
     lot_id = fields.Many2one(
         'stock.production.lot', 'Lot/Serial',
         states={'done': [('readonly', True)]}, domain="[('product_id', '=', product_id), ('company_id', '=', company_id)]", check_company=True)
